src/vs/workbench/browser/layout.ts

```diff
@@ -909,8 +909,28 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 					editorsByGroup.add(editor.editor);
 				}
 
-				openEditorsPromise = Promise.all(Array.from(mapEditorsToGroup).map(async ([groupId, editors]) => {
+				openEditorsPromise = Promise.all(Array.from(mapEditorsToGroup).map(async ([groupId, editorSet]) => {
```

Comment:
@lramos15 can you review this? I'm not an expert in this area

Reply:
I wonder if we could use the `IEditorResolverService` at all here for handling the extnesion activation parts. In this case it seems to be looking through and duplicating a lot of the extension activation stuff based on package.json

Reply:
This is for ensuring that all custom editors contributed by local extensions are registered before we open any editors. I am not quite familiar with this part and this is the best I can figure out. If there's a neater implementation please enlight some instructions.

Reply:
After some research, there seems no way for IEditorResolverService or IExtensionsService to reliably determine whether the custom editors have been registered. Thus it would be necessary to raise activation events and wait on them.

It's worth nothing that even without these activation events, corresponding extensions are also being activated in the background, except that we have no way to know their finish time. The activation events only play a role of finish time notifiers.